core/domain/feedback_jobs_one_off.py

```diff
@@ -100,3 +103,63 @@ def map(item):
     @staticmethod
     def reduce(key, value):
         pass
+
+
+class SuggestionMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for converting all suggestions from the old model to the
+    new model.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.SuggestionModel]
+
+    @staticmethod
+    def map(suggestion):
+        suggestion_id = (
+            suggestion_models.TARGET_TYPE_EXPLORATION + '.' + suggestion.id)
+        thread = feedback_models.FeedbackThreadModel.get_by_id(suggestion.id)
+        if thread.status == feedback_models.STATUS_CHOICES_OPEN:
+            status = suggestion_models.STATUS_RECEIVED
+        elif thread.status == feedback_models.STATUS_CHOICES_FIXED:
+            status = suggestion_models.STATUS_ACCEPTED
+        elif thread.status == feedback_models.STATUS_CHOICES_IGNORED:
+            status = suggestion_models.STATUS_REJECTED
+
+        score_category = (
+            suggestion_models.SCORE_TYPE_CONTENT +
+            suggestion_models.SCORE_CATEGORY_DELIMITER +
+            exp_services.get_exploration_by_id(
+                suggestion.exploration_id).category)
+
+        # For old accepted/rejected suggestions, the audio_translations field
+        # below value is not derivable as it is set only when the
+        # suggestion is accepted and that exploration version is unknown to us.
```

Comment:
This statement seems to contradict the earlier statement or maybe my inference is incorrect. If the field is set when a suggestion is accepted, then, an accepted suggestion should be having it?

Reply:
Ok. So in the old system, the suggestion object had a suggestion_html field which held just the html part (without translations). When accepting the suggestion, the change_list is dynamically created (not stored). When the change_list is created, the audio translations are retrieved from the exploration data and then added to the change cmd. This generated cmd is exactly what we want to store in the new model. But we cannot obtain what the translations were before this suggestion was accepted. So I passed in {}.

For suggestions which are still "in review" in the old system, when they are accepted, the

Reply:
Ok. But why are we concerned with the audio in the suggestions in the first place for this migration job? Sorry for being slow on the uptake , but, I'm not able to understand why audio translations matter when there are no suggestions for the audio translations currently? Also, AFAIK `get_change_list_for_accepting_suggestion()` in the `SuggestionEditStateContent` isn't concerned with pulling the latest audio translations nor does the content contains the audio-translations. Therefore, I'm not sure which scenario we are talking about when we say "...the audio translations are retrieved from the exploration data and then added to the change cmd". Is this regarding the `SuggestionEditAudioTranslation` that we might add in future?

Reply:
Actually no this doesn't relate to the other type of suggestions. The changelist (change_cmd) for this command takes a subtitledHtml object which has both html and audio. This audio is actually retrieved on the fly by get_change_list_for_accepting_suggestion. But actually for any suggestion (of this type) we make, we just need the html and the get_change_list_for_accepting_suggestion will make sure that only content html is updated and translations are not affected. Also even while working on the frontend PR I thought it is better if we just pass html to the backend (as translations need not come into the picture).